src/segment/filter/mod.rs

```diff
@@ -38,3 +45,60 @@ impl BloomConstructionPolicy {
         }
     }
 }
+
+#[enum_dispatch::enum_dispatch]
+pub trait AMQFilter {
+    fn bytes(&self) -> &[u8];
+    fn len(&self) -> usize;
+    fn contains(&self, item: &[u8]) -> bool;
+    fn contains_hash(&self, hash: (u64, u64)) -> bool;
+}
+
+#[enum_dispatch::enum_dispatch(AMQFilter)]
+#[derive(PartialEq, Debug)]
+pub enum BloomFilter {
+    StandardBloom(StandardBloomFilter),
+    BlockedBloom(BlockedBloomFilter),
+}
+
+pub enum BloomFilterType {
```

Comment:
Can't think of a way to not introduce this enum. Could be missing something.

The primary purpose of this is to have a enumeration of filter_type values during decode & encode.

Reply:
Yeah we need this